specifyweb/frontend/js_src/lib/components/Formatters/Formatter.tsx

```diff
@@ -145,51 +147,142 @@ function Definitions({
     }
   }, [needFormatter, handleChange, formatter.definition.fields]);
 
+  const [showConditionalField, setShowConditionalField] = React.useState(
+    trimmedFields.map(() => false)
+  );
+
   return table === undefined ? null : (
     <div className="flex flex-col gap-4 divide-y divide-gray-500 [&>*]:pt-4">
       {trimmedFields.map(({ value, fields }, index) => (
-        <div className="flex flex-col gap-2" key={index}>
+        <div
```

Comment:
turn each row into a component, and then have `isCollapsed` state inside of it instead of `showConditionalField[index]`

a general idea - instead of looping over the logic for each child in the parent, the parent should just render the children, and each child handles it's own state and logic as much as possible - this simplifies the code usually and avoids bugs.

for example, you currently have a bug:
- if a row is deleted, you forget to also delete from showConditionalField
- also, if formatter.definition where to be modified by some other component, your showConditionalField would become stale

Reply:
I gave it a try let me know how it is

Reply:
good job creating ConditionalFormatter

now, can you also remove showConditionalField?
and replace that with a simple boolean state that would be inside of ConditionalFormatter?
and that way, you won't have to manage showConditionalField at all - because if ConditionalFormatter is removed or added, its state would be removed or created automatically
see how that would be simpler, and less bug-prone?

or did you encounter problems implementing this?